### PR TITLE
[MI-3810]: Disconnect API and Toast integration

### PR DESCRIPTION
**Pull request by SaurabhSharma-884**
## Ticket ID
[MI-3810](https://brightscout.atlassian.net/browse/MI-3810)

## Description
- Create Dialog component with hook
- Integrate Snackbar(Toast) wherever required
- Connected User View + Disconnect user API integration (websocket integration)
- Handle loading state for RHs

## Screenshots
https://github.com/Brightscout/mattermost-plugin-msteams-sync/assets/115625377/5e6f0ccc-e2b1-4119-b026-20cba6457215

### CODE REVIEW

File: webapp/src/services/index.ts [hunk 1]
```diff
@@ -43,5 +43,12 @@ export const msTeamsPluginApi = createApi({
                 params: {...params},
             }),
         }),
+        [pluginApiServiceConfigs.disconnectUser.apiServiceName]: builder.query<string, APIRequestPayload>({
+            query: () => ({
+                url: pluginApiServiceConfigs.disconnectUser.path,
+                method: pluginApiServiceConfigs.disconnectUser.method,
+                responseHandler: (res) => res.text(),
```
Comment (ayusht2810) [inline webapp/src/services/index.ts:50]: are we using this response anywhere?

File: webapp/src/types/common/index.d.ts [hunk 2]
```diff
@@ -10,6 +10,11 @@ type ConnectData = {
     connectUrl: string;
 }
 
+type WebsocketEventParams = {
```
Comment (ayusht2810) [inline webapp/src/types/common/index.d.ts:13]: this is defined in the file below as well

**Review by ayusht2810 — APPROVED** [review]

File: webapp/src/containers/Rhs/views/ConnectedAccount/ConnectedAccount.container.tsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        },
+    });
+
+    useApiRequestCompletionState({
+        serviceName: pluginApiServiceConfigs.disconnectUser.apiServiceName,
+        handleSuccess: () => {
+            dispatch(setConnected({connected: false, username: '', msteamsUserId: '', isAlreadyConnected: false}));
+            hideDialog();
+            showAlert({
+                message: 'Your account has been disconnected',
```
Comment (avas27JTG) [inline webapp/src/containers/Rhs/views/ConnectedAccount/ConnectedAccount.container.tsx:40]:
```suggestion
                message: 'Your account has been disconnected.',
```

File: webapp/src/containers/Rhs/views/ConnectedAccount/ConnectedAccount.container.tsx [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+                message: 'Error occurred while disconnecting the user.',
+                severity: 'error',
+            });
+            hideDialog();
+        },
+    });
+
+    useEffect(() => {
+        if (connected && !isAlreadyConnected) {
+            showAlert({message: 'Your account is connected successfully'});
```
Comment (avas27JTG) [inline webapp/src/containers/Rhs/views/ConnectedAccount/ConnectedAccount.container.tsx:54]:
```suggestion
            showAlert({message: 'Your account is connected successfully.'});
```

**Review by avas27JTG — APPROVED** [review]

**Pull request merged**